packages/calypso-storybook/src/index.js

```diff
@@ -0,0 +1,65 @@
+const calypsoBabelConfig = require( '@automattic/calypso-babel-config' );
+
+module.exports = function storybookDefaultConfig( {
+	stories,
+	plugins = [],
+	webpackAliases = {},
+	babelCacheDirectory,
+} = {} ) {
+	return {
+		core: {
+			builder: 'webpack5',
+		},
+		features: {
+			/**
+			 * Can probably be removed after the next major storybook release with emotion 11 support.
+			 *
+			 * @see https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#emotion11-quasi-compatibility
+			 */
+			emotionAlias: false,
+			babelModeV7: true,
+		},
+
+		babel: async ( storybookConfig ) => {
```

Comment:
It's not very well documented, but you can have `babelModeV7: true` and a custom babel config 🤷🏼 .